### PR TITLE
Replicated Event Sourcing support

### PR DESCRIPTION
**Pull request by johanandren**
This won't work until we have released an Akka version with Active Active.

A bit messier than necessary because I kept support for MultiDC but maybe we should throw that away?

### CODE REVIEW

File: core/src/main/scala/akka/persistence/cassandra/snapshot/CassandraSnapshotStore.scala [hunk 1]
```diff
@@ -259,19 +264,30 @@ import akka.stream.alpakka.cassandra.scaladsl.{ CassandraSession, CassandraSessi
...
         // meta data, if any
         payload match {
           case SnapshotWithMetaData(_, m) =>
-            val m2 = m.asInstanceOf[AnyRef]
-            val serializer = serialization.findSerializerFor(m2)
-            val serManifest = Serializers.manifestFor(serializer, m2)
-            val metaBuf = ByteBuffer.wrap(serialization.serialize(m2).get)
-            Some(SerializedMeta(metaBuf, serManifest, serializer.identifier))
-          case _ => None
+            // FIXME can we remove support for the commercial version of this?
```
Comment (chbatey) [inline core/src/main/scala/akka/persistence/cassandra/snapshot/CassandraSnapshotStore.scala:280]: May not matter but we give precedence the other way around for the events, first we check in the PR then we look at the payload.

File: core/src/main/scala/akka/persistence/cassandra/Extractors.scala [hunk 2]
```diff
@@ -175,14 +177,19 @@ import akka.persistence.query.TimeBasedUUID
...
           payload,
           sequenceNr = row.getLong("sequence_nr"),
           persistenceId = row.getString("persistence_id"),
           manifest = row.getString("event_manifest"), // manifest for event adapters
           deleted = false,
           sender = null,
           writerUuid = row.getString("writer_uuid"))
+
+        payload match {
+          case EventWithMetaData(p, m: ReplicatedEventMetadata) => repr.withMetadata(m).withPayload(p)
```
[the remark this answers is not in the document]
Comment (johanandren) [reply core/src/main/scala/akka/persistence/cassandra/Extractors.scala:190]: If it is not ReplicatedEventMetadata it is the "old" Multi-DC scheme where we return the wrapped event

[the remark this answers is not in the document]
Comment (patriknw) [reply core/src/main/scala/akka/persistence/cassandra/Extractors.scala:190]: Multi-DC Persistence doesn't work together with the akka-persistence-cassandra 1.0.x anyway, so it should be safe to remove.

File: core/src/main/scala/akka/persistence/cassandra/snapshot/CassandraSnapshotStore.scala [hunk 3]
```diff
@@ -259,19 +264,30 @@ import akka.stream.alpakka.cassandra.scaladsl.{ CassandraSession, CassandraSessi
...
+            Some(toSerializedForm(m))
+          case _ =>
+            metadata.metadata match {
+              case Some(aaMetadata) =>
+                // active active metadata
```
[the remark this answers is not in the document]
Comment (patriknw) [reply core/src/main/scala/akka/persistence/cassandra/snapshot/CassandraSnapshotStore.scala:285]: replicated

**Comment by patriknw** [comment]
ping @johanandren

**Comment by johanandren** [comment]
Messed up, this is superseded by #797

**Pull request closed without merging**